addons/docs/jest-transform.js

```diff
@@ -0,0 +1,20 @@
+const mdx = require('@mdx-js/mdx');
+const babel = require('babel-jest');
+const deasyncPromise = require('deasync-promise');
+const createCompiler = require('./mdx-compiler-plugin');
+
+const compilers = [createCompiler({})];
+
+module.exports = {
+  process(src, filename, config, options) {
+    let result = deasyncPromise(mdx(src, { compilers, filepath: filename }));
+
+    result = `/* @jsx mdx */
+    import React from 'react'
+    import { mdx } from '@mdx-js/react'
+    ${result}
+    `;
+
+    return babel.process(result, filename, config, options);
```

Comment:
It won't work with custom babel-jest transformer (like the one we [have]() in our own repo),  you may want to borrow the approach from [`injectFileName`](https://github.com/storybookjs/storybook/blob/38d7464d2b2db0b36832b70030ea6c815ac72e7d/addons/storyshots/storyshots-core/injectFileName.js) instead